src/client/src/orangehrmRecruitmentPlugin/components/VacancyCard.vue

```diff
@@ -0,0 +1,181 @@
+        <pre v-if="vacancyDescription" class="orangehrm-vacancy-card-pre-tag">
+          {{ vacancyDescription }}
+        </pre>
+      </oxd-text>
+    </div>
+    <div
+      v-if="vacancyDescription && vacancyDescription.length > descriptionLength"
+      class="orangehrm-vacancy-card-footer"
+    >
+      <a @click="viewDetails">
+        <oxd-text
+          class="orangehrm-vacancy-card-anchor-tag"
+          type="toast-message"
+        >
+          {{
+            isViewDetails
+              ? $t('recruitment.show_more')
+              : $t('recruitment.show_less')
+          }}
+        </oxd-text>
+      </a>
+    </div>
+  </div>
+</template>
+
+<script>
+import OxdDivider from '@ohrm/oxd/core/components/Divider/Divider';
+import useResponsive from '@ohrm/oxd/composables/useResponsive';
+import {toRefs} from 'vue';
+
+export default {
+  name: 'VacancyCard',
+  components: {
+    'oxd-divider': OxdDivider,
+  },
+  props: {
+    vacancyId: {
+      type: Number,
+      required: true,
+    },
+    vacancyTitle: {
+      type: String,
+      required: true,
+    },
+    vacancyDescription: {
+      type: String,
+      required: true,
+    },
+  },
+  setup() {
+    const responsiveState = useResponsive();
+    return {
+      ...toRefs(responsiveState),
+    };
+  },
+  data() {
+    return {
+      viewMore: true,
+      descriptionLength: 0,
+    };
+  },
+  computed: {
+    isViewDetails() {
+      return this.viewMore;
+    },
+    isMobile() {
+      return this.windowWidth < 600;
+    },
+  },
+  watch: {
+    isMobile: function() {
+      this.validDescriptionLength();
+    },
+  },
+  beforeMount() {
+    this.validDescriptionLength();
+  },
+  methods: {
+    viewDetails() {
+      return (this.viewMore = !this.viewMore);
+    },
+    validDescriptionLength() {
+      return this.isMobile
+        ? (this.descriptionLength = 150)
+        : this.windowWidth < 1920
+        ? (this.descriptionLength = 250)
+        : (this.descriptionLength = 400);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+@import '@ohrm/oxd/styles/_mixins.scss';
+
+.orangehrm-paper-container {
+  width: 80%;
+  margin: 1rem auto;
+  padding: 0.75rem 1.5rem;
+}
+
+.orangehrm-vacancy-card {
+  display: block;
+
+  &-header {
+    display: flex;
+    justify-content: space-between;
+  }
+
+  &-anchor-tag {
+    color: $oxd-primary-one-color;
+    padding-top: 0.5rem;
+  }
+
+  &-anchor-tag:hover {
+    cursor: pointer;
+  }
+
+  &-body {
+    text-align: left;
+    justify-content: space-between;
+    word-break: break-all;
+    max-height: 60px;
+    -webkit-line-clamp: 4;
+    @include oxd-respond-to('md') {
+      max-height: 30px;
+      -webkit-line-clamp: 2;
+    }
+    text-overflow: ellipsis;
+    overflow: hidden;
+    display: -webkit-box;
+    -webkit-box-orient: vertical;
+  }
+  &-pre-tag {
+    margin-top: 0;
+    font-family: $oxd-font-family;
+    white-space: pre-wrap;
+    white-space: -moz-pre-wrap;
```

Comment:
remove

Reply:
if I remove it, formatted text view breaks.